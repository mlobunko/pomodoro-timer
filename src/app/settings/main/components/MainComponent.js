import React from "react";
import PropTypes from "prop-types";

import Button from "shared/button/components";

import Work from "../../work/components";
import Rest from "../../rest/components";
import BigRest from "../../bigRest/components";
import Sound from "../../sound/components";

import {
  StyledButtons,
  StyledCenterOfPage,
  StyledHeader,
  StyledListOfSettings,
  StyledMainComponent
} from "./styles";

export const MainComponent = ({ closeSettings }) => (
  <StyledMainComponent>
    <StyledHeader>Settings</StyledHeader>
    <StyledCenterOfPage>
      <StyledListOfSettings>
        <Work />
        <Rest />
        <BigRest />
        <Sound />
      </StyledListOfSettings>
    </StyledCenterOfPage>
    <StyledButtons>
      <Button onClick={closeSettings} />
    </StyledButtons>
  </StyledMainComponent>
);

export default MainComponent;

MainComponent.displayName = "SettingsMain";

MainComponent.propTypes = {
  closeSettings: PropTypes.func.isRequired
};
