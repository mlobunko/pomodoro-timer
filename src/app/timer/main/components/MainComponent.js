import React from "react";

import Buttons from "../../buttons/components";
import Header from "../../header/components";
import Tomato from "../../tomato/components";
import LittleTomatoes from "../../littleTomatoes/components";
import {
  StyledBackground,
  StyledCenterOfPage,
  StyledFront,
  StyledPage
} from "./styles";

export const MainComponent = () => (
  <React.Fragment>
    <StyledBackground />
    <StyledFront>
      <StyledPage>
        <Header />
        <StyledCenterOfPage>
          <Tomato />
          <LittleTomatoes />
        </StyledCenterOfPage>
        <Buttons />
      </StyledPage>
    </StyledFront>
  </React.Fragment>
);

MainComponent.displayName = "TimerMain";

export default MainComponent;
