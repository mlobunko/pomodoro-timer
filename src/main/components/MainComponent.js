import React from "react";

import SettingsMain from "app/settings/main/components";
import TimerMain from "app/timer/main/components";

import ErrorBoundary from "./ErrorBoundary";
import { StyledGlobal } from "./styles";

export const MainComponent = ({ isSettingsOpen }) => (
  <ErrorBoundary>
    <StyledGlobal>
      {isSettingsOpen ? <SettingsMain /> : <TimerMain />}
    </StyledGlobal>
  </ErrorBoundary>
);

MainComponent.displayName = "Main";

export default MainComponent;
