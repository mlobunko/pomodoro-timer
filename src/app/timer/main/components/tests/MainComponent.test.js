import React from "react";
import { shallow } from "enzyme";

import { MainComponent } from "../MainComponent";

describe("render", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<MainComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
