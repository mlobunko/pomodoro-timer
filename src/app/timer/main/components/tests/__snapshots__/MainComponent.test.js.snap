// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<Fragment>
  <styled.div />
  <styled.div>
    <styled.div>
      <Header />
      <styled.div>
        <Connect(Tomato) />
        <LittleTomatoes />
      </styled.div>
      <Connect(Buttons) />
    </styled.div>
  </styled.div>
</Fragment>
`;
